core/src/main/java/feign/Response.java

```diff
@@ -242,4 +246,19 @@ public String toString() {
       return decodeOrDefault(data, UTF_8, "Binary data");
     }
   }
+
+  private static Map<String, Collection<String>> caseInsensitiveCopyOf(Map<String, Collection<String>> headers) {
```

Comment:
tests would be less crazy, if you make this package protected and test it directly.